fuse/fuse.cpp

```diff
@@ -92,34 +125,15 @@ static int execve_wrapper(const std::vector<std::string>& command, const std::ve
 	return errno;
 }
 
-int run_in_fuse(const fuse_args& args, std::string& result_json) {
-	if (0 != chdir(args.working_dir.c_str())) {
-		std::cerr << "chdir " << args.working_dir << ": " << strerror(errno) << std::endl;
-		return 1;
-	}
-
-	std::string name  = std::to_string(getpid());
-	// mpath is where the fuse filesystem is mounted
-	std::string mpath = args.working_dir + "/.fuse";
-	std::string fpath = mpath + "/.f.fuse-waked";
-	// rpath is a subdir in the fuse filesystem that will be used by this fuse-wake
-	std::string rpath = mpath + "/" + name;
-	// Lock file held by each child of fuse-wake. When all children close it,
-	// the daemon releases the resources for that job
-	std::string lpath = mpath + "/.l." + name;
-	// Input (as json) to the fuse daemon to setup visible files.
-	std::string ipath = mpath + "/.i." + name;
-	// Result metadata from the daemon
-	std::string opath = mpath + "/.o." + name;
-
+bool contact_daemon(struct daemon_locations &dls, int &livefd, std::vector<std::string> &visible){
```

Comment:
please rename dls to daemon, then `daemon.is_running` and `daemon.executable` all make sense to read